From 720e5edce6faf38aa52bc3eecae1945c2f6675dc Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Tue, 17 Nov 2015 04:25:41 +0200
Subject: [PATCH 0507/1240] armada3700: comphy: Fix the dedicated PHYs init

- Remove code looking for subnodes in DT for USB/SATA/SDIO
  nodes. These nodes have no subnodes at all.
- Fix compat values for SATA and SDIO
- Add more debug prints

Change-Id: Ied6531d241c31de1e7ef8779b3f13b58696ad237
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/24943
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/phy/comphy_a3700.c | 62 ++++++++++++++++++++++++++--------------------
 1 file changed, 35 insertions(+), 27 deletions(-)

diff --git a/drivers/phy/comphy_a3700.c b/drivers/phy/comphy_a3700.c
index 539ba2c..1dbad52 100644
--- a/drivers/phy/comphy_a3700.c
+++ b/drivers/phy/comphy_a3700.c
@@ -633,7 +633,7 @@ static int comphy_sgmii_power_up(u32 lane)
  ***************************************************************************************************/
 static void comphy_dedicated_phys_init(void)
 {
-	int node, subnode, count, ret = 1;
+	int node, count, ret = 1;
 	const void *blob = gd->fdt_blob;
 
 	debug_enter();
@@ -642,40 +642,48 @@ static void comphy_dedicated_phys_init(void)
 			COMPAT_MVEBU_USB, &node, A3700_MAX_USB_CNT);
 
 	if (count > 0) {
-		fdt_for_each_subnode(blob, subnode, node) {
-			if (fdtdec_get_is_enabled(blob, subnode)) {
-				ret = comphy_usb2_power_up();
-				if (ret == 0)
-					error("Failed to initialize USB2 PHY\n");
-			}
-		}
-	}
+		if (fdtdec_get_is_enabled(blob, node)) {
+			ret = comphy_usb2_power_up();
+			if (ret == 0)
+				error("Failed to initialize USB2 PHY\n");
+			else
+				debug("USB PHY init succeed\n");
+		} else
+			debug("USB node is disabled\n");
+	} else
+		debug("No USB node in DT\n");
 
 	count = fdtdec_find_aliases_for_id(blob, "sata",
-			COMPAT_MVEBU_USB, &node, A3700_MAX_SATA_CNT);
+			COMPAT_MVEBU_SATA, &node, A3700_MAX_SATA_CNT);
 
 	if (count > 0) {
-		fdt_for_each_subnode(blob, subnode, node) {
-			if (fdtdec_get_is_enabled(blob, subnode)) {
-				ret = comphy_sata_power_up();
-				if (ret == 0)
-					error("Failed to initialize SATA PHY\n");
-			}
-		}
-	}
+		if (fdtdec_get_is_enabled(blob, node)) {
+			ret = comphy_sata_power_up();
+			if (ret == 0)
+				error("Failed to initialize SATA PHY\n");
+			else
+				debug("SATA PHY init succeed\n");
+		} else
+			debug("SATA node is disabled\n");
+	}  else
+		debug("No SATA node in DT\n");
+
 
 	count = fdtdec_find_aliases_for_id(blob, "sdio",
-		 COMPAT_MVEBU_USB, &node, A3700_MAX_SDIO_CNT);
+		 COMPAT_MVEBU_SDIO, &node, A3700_MAX_SDIO_CNT);
 
 	if (count > 0) {
-		fdt_for_each_subnode(blob, subnode, node) {
-			if (fdtdec_get_is_enabled(blob, subnode)) {
-				ret = comphy_emmc_power_up();
-				if (ret == 0)
-					error("Failed to initialize SDIO/eMMC PHY\n");
-			}
-		}
-	}
+		if (fdtdec_get_is_enabled(blob, node)) {
+			ret = comphy_emmc_power_up();
+			if (ret == 0)
+				error("Failed to initialize SDIO/eMMC PHY\n");
+			else
+				debug("SDIO/eMMC PHY init succeed\n");
+		} else
+			debug("SDIO/eMMC node is disabled\n");
+	}  else
+		debug("No SDIO/eMMC node in DT\n");
+
 
 	debug_exit();
 }
-- 
1.9.1

